internal/provider/data_host.go

```diff
@@ -0,0 +1,473 @@
+// © Broadcom. All Rights Reserved.
+// The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vcf-sdk-go/client"
+	"github.com/vmware/vcf-sdk-go/client/hosts"
+	"github.com/vmware/vcf-sdk-go/models"
+
+	"github.com/vmware/terraform-provider-vcf/internal/api_client"
+	"github.com/vmware/terraform-provider-vcf/internal/constants"
+)
+
+func DataSourceHost() *schema.Resource {
```

Comment:
Stoyo implemented a hybrid old-new mechanism for this provider. Isn't it better for new resources to be implemented using the new "the framework" as Hashicorp call it, instead of the old SDK?

Reply:
Done! Just need to test in an environment.